frontend/src/container/GridCardLayout/GridCardLayout.tsx

```diff
@@ -0,0 +1,141 @@
+import { useUpdateDashboard } from 'hooks/dashboard/useUpdateDashboard';
+import useComponentPermission from 'hooks/useComponentPermission';
+import { useIsDarkMode } from 'hooks/useDarkMode';
+import { useNotifications } from 'hooks/useNotifications';
+import { useDashboard } from 'providers/Dashboard/Dashboard';
+import { useSelector } from 'react-redux';
+import { AppState } from 'store/reducers';
+import { Dashboard, Widgets } from 'types/api/dashboard/getAll';
+import AppReducer from 'types/reducer/app';
+
+import GridCard from './GridCard';
+import {
+	Button,
+	ButtonContainer,
+	Card,
+	CardContainer,
+	ReactGridLayout,
+} from './styles';
+import { GraphLayoutProps } from './types';
+import { MenuItemKeys } from './WidgetHeader/contants';
+
+function GraphLayout({
+	onAddPanelHandler,
+	widgets,
+}: GraphLayoutProps): JSX.Element {
+	const { selectedDashboard, layouts, setLayouts } = useDashboard();
+
+	const { featureResponse } = useSelector<AppState, AppReducer>(
+		(state) => state.app,
+	);
+
+	const { role } = useSelector<AppState, AppReducer>((state) => state.app);
+
+	const isDarkMode = useIsDarkMode();
+
+	const updateDashboardMutation = useUpdateDashboard();
+
+	const { notifications } = useNotifications();
+
+	const [saveLayoutPermission, addPanelPermission] = useComponentPermission(
+		['save_layout', 'add_panel'],
+		role,
+	);
+
+	const onSaveHandler = (): void => {
+		if (!selectedDashboard) return;
+
+		const { data } = selectedDashboard;
+
+		const updatedDashboard: Dashboard = {
+			...selectedDashboard,
+			data: {
+				title: data.title,
+				description: data.description,
+				name: data.name,
+				tags: data.tags,
+				widgets: data.widgets,
+				variables: data.variables,
+				layout: layouts.filter((e) => e.i !== PANEL_TYPES.EMPTY_WIDGET),
+			},
+			uuid: selectedDashboard.uuid,
+		};
+
+		updateDashboardMutation.mutate(updatedDashboard, {
+			onSuccess: () => {
+				featureResponse.refetch();
+
+				notifications.success({
+					message: 'Layout saved successfully',
+				});
+			},
+			onError: () => {
+				notifications.error({
+					message: SOMETHING_WENT_WRONG,
+				});
+			},
+		});
+	};
+
+	return (
+		<>
+			<ButtonContainer>
+				{saveLayoutPermission && (
+					<Button
+						loading={updateDashboardMutation.isLoading}
+						onClick={onSaveHandler}
+						icon={<SaveFilled />}
+						disabled={updateDashboardMutation.isLoading}
+					>
+						Save Layout
+					</Button>
+				)}
+
+				{addPanelPermission && (
+					<Button onClick={onAddPanelHandler} icon={<PlusOutlined />}>
+						Add Panel
+					</Button>
+				)}
+			</ButtonContainer>
+
+			<ReactGridLayout
+				cols={12}
+				rowHeight={100}
+				autoSize
+				width={100}
+				isDraggable={addPanelPermission}
+				isDroppable={addPanelPermission}
+				isResizable={addPanelPermission}
+				allowOverlap={false}
+				onLayoutChange={setLayouts}
+				draggableHandle=".drag-handle"
+			>
+				{layouts.map(({ ...rest }) => {
```

Comment:
Since we are not spreading any other prop here, it can be renamed to 
```suggestion
				{layouts.map((layout)) => {
```
or spread only the required properties
```suggestion
				{layouts.map(({i})) => {
```